onnxruntime/core/common/spin_pause.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#if defined(_M_AMD64)
+#include <intrin.h>
+#endif
+
+#if defined(__x86_64__)
+#include <xmmintrin.h>
+#endif
+
+#if defined(_M_AMD64) || defined(__x86_64__)
+#if defined(__linux__)
+#include <x86intrin.h>
+#include <immintrin.h>
+#include <cstdlib>
+#elif defined(_WIN32)
+#include <Windows.h>
+#endif
+#include <cstdint>
+#include <string>
+#include "core/common/cpuid_info.h"
+#endif
+
+#include "core/common/spin_pause.h"
```

Comment:
nit: this one should be first.

https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes